CHANGELOG.md

```diff
@@ -2,6 +2,7 @@
 
 ### Enhancements
 * <New feature description> (PR [#????](https://github.com/realm/realm-core/pull/????))
+* Report the originating error that caused a client reset to occur. ([#6154](https://github.com/realm/realm-core/issues/6154))
```

Comment:
are we actually surfacing the error? could not find where it's done

Reply:
We do print the error with the rest of the PendingReset structure info:
https://github.com/realm/realm-core/blob/mwb/add-extra-client-reset-fields/src/realm/sync/noinst/pending_reset_store.cpp#L33-L49

Which will be provided via debug out here:
https://github.com/realm/realm-core/blob/mwb/add-extra-client-reset-fields/src/realm/sync/client.cpp#L1991

Reply:
Example:
```
Realm.Sync.Client.Reset - Connection[1] Session[5]: Tracking pending client reset of type: 'Recover' at: 2024-05-31 00:38:03.146563000 for error: SyncClientResetRequired: Bad client file identifier (IDENT)
. . .
Realm.Sync.Client.Reset - Connection[1] Session[5]: Server has acknowledged pending client reset of type: 'Recover' at: 2024-05-31 00:38:03.146563000 for error: SyncClientResetRequired: Bad client file identifier (IDENT)
```

Reply:
what I meant is surfacing it through a SyncError in handle_error so the users know what caused the initial client reset.